tests/src/testcommon.hh

```diff
@@ -278,8 +259,181 @@ template <typename NonLinearOperator, typename FiniteElement>
+  auto computedResult = result.getResult(ResultType::linearStress);
+
+  for (const auto vI : std::views::iota(0u, 6u)) {
+    const auto feStressComponent       = computedResult(vI, 0);
+    const auto expectedStressComponent = expectedStress[vI];
+    const bool isStressCorrect
+        = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(feStressComponent, expectedStressComponent);
+    t.check(isStressCorrect) << "Stress component " << vI << " is not correctly computed" << messageIfFailed;
+  }
+
+  return t;
+}
+
+template <typename NonLinearOperator, typename FiniteElement>
+[[nodiscard]] auto checkLinearStressOf2DSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                                const std::string& messageIfFailed = "") {
+  static_assert(FiniteElement::Traits::mydim == 2,
+                "The test to check linear stress is only supported for the three-dimensional case");
+  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(2)
+         && "The test to check linear stress is only supported for 2d simplexes (tetrahedrons)");
+  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == 2,
+                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
+  static_assert(std::is_same_v<typename std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
+                               Dune::Functions::LagrangeNode<
+                                   std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1, double>>,
+                "The test to check linear stress is only supported for a linear Lagrange basis");
+
+  Dune::TestSuite t("Linear Stress check of 2D solid element (3-node triangle)");
+
+  using namespace Ikarus;
+  using namespace Dune::Indices;
+  using namespace Dune::Functions::BasisFactory;
+
+  auto& displacement = nonLinearOperator.firstParameter();
+  displacement << 0, 0, 2, 0, 1, 0;
+
+  Dune::FieldVector<double, 3> expectedStress = {2197.8021978, 659.3406593, 384.61538461};
+
+  auto resultRequirements = Ikarus::ResultRequirements<>()
+                                .insertGlobalSolution(Ikarus::FESolutions::displacement, displacement)
+                                .addResultRequest(ResultType::linearStress);
+  auto gridView        = fe.localView().globalBasis().gridView();
+  auto scalarBasis     = makeConstSharedBasis(gridView, lagrangeDG<1>());
+  auto localScalarView = scalarBasis->localView();
+
+  ResultTypeMap result;
+
+  auto ele = elements(gridView).begin();
+  localScalarView.bind(*ele);
+  const auto& fe2 = localScalarView.tree().finiteElement();
+
+  // Constant stress state for linear element
+  fe.calculateAt(resultRequirements, Dune::FieldVector<double, 2>{0, 0}, result);
+  auto computedResult = result.getResult(ResultType::linearStress);
+  for (const auto vI : std::views::iota(0u, 3u)) {
```

Comment:
Same comments as for the 3D case. It would be nice, if these tests can be merged into one, and then it really can be called checkLinearStressOfSimplex and internally depending on the dimension of `FiniteElement` decided which results should be checked

Reply:
For now I ll merge the two simlpex tests into one, we can then discuss if there should be a more generel `checkStress` Function that can handle all type of elements

Reply:
Also i think we should split the functionality in `testcommon.hh` a bit. 
E.g. a file for 
1. Create Grid Function( i think there is one for ALU, YASP and IGA)
2. generel test functions
3. stress tests
4. utility tests etc...

what do you think? @rath3t @tarun-mitruka

Reply:
Yes this should be seperated. Be also aware that there is a mechanism that adds a specific test for _your_ very element: https://github.com/ikarus-project/ikarus/blob/35fc5a1b3c9c9dbb2063e6a8e5b49feadd3a2dc9/tests/src/testfeelement.hh#L112C28-L114

One implementation for this resides at https://github.com/ikarus-project/ikarus/blob/35fc5a1b3c9c9dbb2063e6a8e5b49feadd3a2dc9/tests/src/testeas.hh#L13 but IMHO this could also be in the header of the element itself.

Thus in the test folder there are other headers which test for generic stuff. But then there should be special tests for your element, where you really provide numbers for your element